tests/bluetooth/tester/src/btp_ccp.c

```diff
@@ -871,3 +872,304 @@ uint8_t tester_unregister_ccp(void)
 {
 	return BTP_STATUS_SUCCESS;
 }
+
+/* Telephone Bearer Service */
+static uint8_t tbs_supported_commands(const void *cmd, uint16_t cmd_len, void *rsp,
+				       uint16_t *rsp_len)
+{
+	struct btp_tbs_read_supported_commands_rp *rp = rsp;
+
+	/* octet 0 */
+	tester_set_bit(rp->data, BTP_TBS_READ_SUPPORTED_COMMANDS);
+	tester_set_bit(rp->data, BTP_TBS_REMOTE_INCOMING);
+	tester_set_bit(rp->data, BTP_TBS_HOLD);
+	tester_set_bit(rp->data, BTP_TBS_SET_BEARER_NAME);
+	tester_set_bit(rp->data, BTP_TBS_SET_TECHNOLOGY);
+	tester_set_bit(rp->data, BTP_TBS_SET_URI_SCHEME);
+	tester_set_bit(rp->data, BTP_TBS_SET_STATUS_FLAGS);
+
+	/* octet 1 */
+	tester_set_bit(rp->data, BTP_TBS_REMOTE_HOLD);
+	tester_set_bit(rp->data, BTP_TBS_ORIGINATE);
+	tester_set_bit(rp->data, BTP_TBS_SET_SIGNAL_STRENGTH);
+
+	*rsp_len = sizeof(*rp) + 1;
+
+	return BTP_STATUS_SUCCESS;
+}
+
+static uint8_t tbs_remote_incoming(const void *cmd, uint16_t cmd_len, void *rsp, uint16_t *rsp_len)
+{
+	const struct btp_tbs_remote_incoming_cmd *cp = cmd;
+	char friendly_name[CONFIG_BT_TBS_MAX_URI_LENGTH];
+	char caller_uri[CONFIG_BT_TBS_MAX_URI_LENGTH];
+	char rcv_uri[CONFIG_BT_TBS_MAX_URI_LENGTH];
+	char uri_string[cp->data_len];
+	int err;
+
+	LOG_DBG("");
+
+	if ((cp->recv_len >= sizeof(rcv_uri) || cp->caller_len >= sizeof(rcv_uri)) ||
+	     cp->fn_len >= sizeof(friendly_name)) {
+		return BTP_STATUS_FAILED;
+	}
+
+	memcpy(uri_string, cp->data, cp->data_len);
+	uri_string[cp->data_len] = '\0';
+
+	strncpy(rcv_uri, uri_string, cp->recv_len);
+	strncpy(caller_uri, uri_string + cp->recv_len, cp->caller_len);
+	strncpy(friendly_name, uri_string + cp->recv_len + cp->caller_len, cp->fn_len);
```

Comment:
Suggest to add checks for whether they can fit in the supplied arrays first.

Reply:
This applies for all of these functions that handle variable length input

Reply:
This one is verified by:

if ((cp->recv_len >= sizeof(recv_uri) || cp->caller_len >= sizeof(recv_uri)) ||
      cp->fn_len >= sizeof(friendly_name)) {
		return BTP_STATUS_FAILED;
	}

And it's done the same way in other similar functions. What else should I verify?